Import generates invalid HCL for map key with period

### Community note

I was recently trying to move from render.com's proprietary YAML IaC format to TF. I generated the import configuration and ran `tofu plan -generate-config-out=render-generated.tf`. However I found that in this case the some keys are added that are in an invalid format:

```hcl
resource "render_env_group" "foo" {
  environment_id = "evm-phngluimglwnafh"
  name           = "foo"
  secret_files = {
    config.json = {
      content = "{}"
    }
  }
}
```

In this case the key `config.json` is not valid and the plan will fail. I had to do a post-processing pass that quoted such keys to get the TF file to work:

```js
renderGeneratedTf.replace(/\b([^"\s]+(?:\..+)+) = /g, (_, fileKey) => `"${fileKey}" = `);
```
>[!NOTE]
>This pass specifically fixes periods and will for example not fix keys with spaces

If someone could point me to a way to reproduce this using a mock, I'd gladly provide a reproduction.

### OpenTofu Version

```shell
OpenTofu v1.9.0
on linux_amd64
+ provider registry.terraform.io/render-oss/render v1.4.0
```

### OpenTofu Configuration Files

```hcl
terraform {
  required_providers {
    render = {
      source = "registry.terraform.io/render-oss/render"
      version = "1.4.0"
    }
  }
}

variable "render_api_key" {
  type = string
  sensitive = true
  description = "API key to use when interacting with the API."
}

provider "render" {
  api_key = var.render_api_key
  owner_id = "tea-cthulhurlyehwgahnaglfhtagn"
  skip_deploy_after_service_update = false
  wait_for_deploy_completion = false
}

import {
  to = render_env_group.foo
  id = "evg-phngluimglwnafh"
}

```


### Debug Output

Sorry, this is not the full log, as it leaked some highly sensitive data. I've tried to provide the most relevant, redacted lines:
```log
2025-02-14T09:07:21.593+0100 [WARN]  Provider "registry.terraform.io/render-oss/render" produced an unexpected new value for render_env_group.staging_api during refresh.
      - .name: was null, but now cty.StringVal("foo")
      - .secret_files: was null, but now cty.MapVal(map[string]cty.Value{"config.json":cty.ObjectVal(map[string]cty.Value{"content":cty.StringVal("{}")})})
2025-02-14T09:07:21.596+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache
2025-02-14T09:07:21.596+0100 [TRACE] NodeAbstractResourceInstance.writeResourceInstanceState to refreshState for render_env_group.foo
2025-02-14T09:07:21.596+0100 [TRACE] NodeAbstractResourceInstance.writeResourceInstanceState: writing state object for render_env_group.foo
2025-02-14T09:07:21.597+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache
2025-02-14T09:07:21.597+0100 [TRACE] NodeAbstractResourceInstance.writeResourceInstanceState to prevRunState for render_env_group.foo
2025-02-14T09:07:21.597+0100 [TRACE] NodeAbstractResourceInstance.writeResourceInstanceState: writing state object for render_env_group.foo
2025-02-14T09:07:21.597+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache
2025-02-14T09:07:21.597+0100 [TRACE] NodeAbstractResourceInstance.writeResourceInstanceState to refreshState for render_env_group.foo
2025-02-14T09:07:21.597+0100 [TRACE] NodeAbstractResourceInstance.writeResourceInstanceState: writing state object for render_env_group.foo
2025-02-14T09:07:21.598+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache
2025-02-14T09:07:21.599+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache
2025-02-14T09:07:21.599+0100 [TRACE] writeChange: recorded NoOp change for render_env_group.foo
2025-02-14T09:07:21.599+0100 [ERROR] vertex "render_env_group.foo" error: Reference to undeclared resource
2025-02-14T09:07:21.599+0100 [TRACE] vertex "render_env_group.foo": visit complete, with errors
2025-02-14T09:07:21.600+0100 [TRACE] dag/walk: upstream of "root" errored, so skipping
2025-02-14T09:07:21.600+0100 [TRACE] vertex "render_env_group.foo (expand)": dynamic subgraph encountered errors: Reference to undeclared resource
2025-02-14T09:07:21.600+0100 [ERROR] vertex "render_env_group.foo (expand)" error: Reference to undeclared resource
2025-02-14T09:07:21.600+0100 [TRACE] vertex "render_env_group.foo (expand)": visit complete, with errors
2025-02-14T09:07:21.600+0100 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/render-oss/render\"] (close)" errored, so skipping
2025-02-14T09:07:21.600+0100 [TRACE] dag/walk: upstream of "root" errored, so skipping
2025-02-14T09:07:21.600+0100 [TRACE] LoadSchemas: retrieving schema for provider type "registry.terraform.io/render-oss/render"
2025-02-14T09:07:21.600+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache
2025-02-14T09:07:21.600+0100 [INFO]  backend/local: plan operation completed
2025-02-14T09:07:21.600+0100 [TRACE] LoadSchemas: retrieving schema for provider type "registry.terraform.io/render-oss/render"
2025-02-14T09:07:21.600+0100 [TRACE] tofu.contextPlugins: Serving provider "registry.terraform.io/render-oss/render" schema from global schema cache

Planning failed. OpenTofu encountered an error while generating this plan.

╷
│ Warning: Config generation is experimental
│ 
│ Generating configuration during import is currently experimental, and the generated configuration format may change in future versions.
╵
╷
│ Error: Reference to undeclared resource
│ 
│   on render-generated.tf line 10:
│   (source code not available)
│ 
│ A managed resource "config" has not been declared in the root module.
```

### Expected Behavior

Expected output file to contain:

```hcl
# __generated__ by OpenTofu
# Please review these resources and move them into your main configuration files.

# __generated__ by OpenTofu
resource "render_env_group" "foo" {
  env_vars = {}
  environment_id = "evm-phngluimglwnafh"
  name           = "foo"
  secret_files = {
    "config.json" = {
      content = null # sensitive
    }
  }
}
```

### Actual Behavior

Output file to contains:

```hcl
# __generated__ by OpenTofu
# Please review these resources and move them into your main configuration files.

# __generated__ by OpenTofu
resource "render_env_group" "foo" {
  env_vars = {}
  environment_id = "evm-phngluimglwnafh"
  name           = "foo"
  secret_files = {
    config.json = {
      content = null # sensitive
    }
  }
}
```

### Steps to Reproduce

1. `tofu init`
2. `tofu plan -generate-config-out=render-generated.tf`

### Additional Context

It sees to be specific to the part: `cty.MapVal(map[string]cty.Value{"config.json":cty.ObjectVal`… when this map is written to the generated HCL, the map key should be quoted, because otherwise the period changes how it's parsed.

### References

_No response_

--- a/internal/genconfig/generate_config.go
+++ b/internal/genconfig/generate_config.go
@@ -12,6 +12,7 @@ import (
 	"strings"
 
 	"github.com/hashicorp/hcl/v2"
+	hclsyntax "github.com/hashicorp/hcl/v2/hclsyntax"
 	"github.com/hashicorp/hcl/v2/hclwrite"
 	"github.com/zclconf/go-cty/cty"
 	"github.com/zclconf/go-cty/cty/json"
@@ -90,6 +91,10 @@ func writeConfigAttributes(addr addrs.AbsResourceInstance, buf *strings.Builder,
 		}
 		if attrS.Required {
 			buf.WriteString(strings.Repeat(" ", indent))
+			// Handle cases where the name should be contained in quotes
+			if !hclsyntax.ValidIdentifier(name) {
+				name = string(hclwrite.TokensForValue(cty.StringVal(name)).Bytes())
+			}
 			buf.WriteString(fmt.Sprintf("%s = ", name))
 			tok := hclwrite.TokensForValue(attrS.EmptyValue())
 			if _, err := tok.WriteTo(buf); err != nil {
@@ -104,6 +109,10 @@ func writeConfigAttributes(addr addrs.AbsResourceInstance, buf *strings.Builder,
 			writeAttrTypeConstraint(buf, attrS)
 		} else if attrS.Optional {
 			buf.WriteString(strings.Repeat(" ", indent))
+			// Handle cases where the name should be contained in quotes
+			if !hclsyntax.ValidIdentifier(name) {
+				name = string(hclwrite.TokensForValue(cty.StringVal(name)).Bytes())
+			}
 			buf.WriteString(fmt.Sprintf("%s = ", name))
 			tok := hclwrite.TokensForValue(attrS.EmptyValue())
 			if _, err := tok.WriteTo(buf); err != nil {

--- a/internal/genconfig/generate_config_test.go
+++ b/internal/genconfig/generate_config_test.go
@@ -768,20 +768,20 @@ resource "tfcoremock_simple_resource" "example" {
 		},
 	}
 	for name, tc := range tcs {
-		t.Run(name, func(tt *testing.T) {
+		t.Run(name, func(t *testing.T) {
 			err := tc.schema.InternalValidate()
 			if err != nil {
-				tt.Fatalf("schema failed InternalValidate: %s", err)
+				t.Fatalf("schema failed InternalValidate: %s", err)
 			}
 			contents, diags := GenerateResourceContents(tc.addr, tc.schema, tc.provider, tc.value)
 			if len(diags) > 0 {
-				tt.Errorf("expected no diagnostics but found %s", diags)
+				t.Errorf("expected no diagnostics but found %s", diags)
 			}
 
 			got := WrapResourceContents(tc.addr, contents)
 			want := strings.TrimSpace(tc.expected)
 			if diff := cmp.Diff(got, want); len(diff) > 0 {
-				tt.Errorf("got:\n%s\nwant:\n%s\ndiff:\n%s", got, want, diff)
+				t.Errorf("got:\n%s\nwant:\n%s\ndiff:\n%s", got, want, diff)
 			}
 		})
 	}

--- a/internal/genconfig/generate_config_test.go
+++ b/internal/genconfig/generate_config_test.go
@@ -84,6 +84,104 @@ resource "tfcoremock_simple_resource" "empty" {
   }
 }`,
 		},
+		"simple_resource_with_propertyname_containing_a_dot": {
+			schema: &configschema.Block{
+				BlockTypes: map[string]*configschema.NestedBlock{
+					"list_block": {
+						Block: configschema.Block{
+							Attributes: map[string]*configschema.Attribute{
+								"nested_value.json": {
+									Type:     cty.String,
+									Optional: true,
+								},
+							},
+						},
+						Nesting: configschema.NestingSingle,
+					},
+				},
+				Attributes: map[string]*configschema.Attribute{
+					"id": {
+						Type:     cty.String,
+						Computed: true,
+					},
+					"value": {
+						Type:     cty.String,
+						Optional: true,
+					},
+				},
+			},
+			addr: addrs.AbsResourceInstance{
+				Module: nil,
+				Resource: addrs.ResourceInstance{
+					Resource: addrs.Resource{
+						Mode: addrs.ManagedResourceMode,
+						Type: "tfcoremock_simple_resource",
+						Name: "empty",
+					},
+					Key: nil,
+				},
+			},
+			provider: addrs.LocalProviderConfig{
+				LocalName: "tfcoremock",
+			},
+			value: cty.NilVal,
+			expected: `
+resource "tfcoremock_simple_resource" "empty" {
+  value = null                 # OPTIONAL string
+  list_block {                 # OPTIONAL block
+    "nested_value.json" = null # OPTIONAL string
+  }
+}`,
+		},
+		"simple_resource_with_propertyname_containing_double_quotes": {
+			schema: &configschema.Block{
+				BlockTypes: map[string]*configschema.NestedBlock{
+					"list_block": {
+						Block: configschema.Block{
+							Attributes: map[string]*configschema.Attribute{
+								"nested_value\"example": {
+									Type:     cty.String,
+									Optional: true,
+								},
+							},
+						},
+						Nesting: configschema.NestingSingle,
+					},
+				},
+				Attributes: map[string]*configschema.Attribute{
+					"id": {
+						Type:     cty.String,
+						Computed: true,
+					},
+					"value": {
+						Type:     cty.String,
+						Optional: true,
+					},
+				},
+			},
+			addr: addrs.AbsResourceInstance{
+				Module: nil,
+				Resource: addrs.ResourceInstance{
+					Resource: addrs.Resource{
+						Mode: addrs.ManagedResourceMode,
+						Type: "tfcoremock_simple_resource",
+						Name: "empty",
+					},
+					Key: nil,
+				},
+			},
+			provider: addrs.LocalProviderConfig{
+				LocalName: "tfcoremock",
+			},
+			value: cty.NilVal,
+			expected: `
+resource "tfcoremock_simple_resource" "empty" {
+  value = null                     # OPTIONAL string
+  list_block {                     # OPTIONAL block
+    "nested_value\"example" = null # OPTIONAL string
+  }
+}`,
+		},
 		"simple_resource_with_state": {
 			schema: &configschema.Block{
 				BlockTypes: map[string]*configschema.NestedBlock{
@@ -670,20 +768,20 @@ resource "tfcoremock_simple_resource" "example" {
 		},
 	}
 	for name, tc := range tcs {
-		t.Run(name, func(t *testing.T) {
+		t.Run(name, func(tt *testing.T) {
 			err := tc.schema.InternalValidate()
 			if err != nil {
-				t.Fatalf("schema failed InternalValidate: %s", err)
+				tt.Fatalf("schema failed InternalValidate: %s", err)
 			}
 			contents, diags := GenerateResourceContents(tc.addr, tc.schema, tc.provider, tc.value)
 			if len(diags) > 0 {
-				t.Errorf("expected no diagnostics but found %s", diags)
+				tt.Errorf("expected no diagnostics but found %s", diags)
 			}
 
 			got := WrapResourceContents(tc.addr, contents)
 			want := strings.TrimSpace(tc.expected)
 			if diff := cmp.Diff(got, want); len(diff) > 0 {
-				t.Errorf("got:\n%s\nwant:\n%s\ndiff:\n%s", got, want, diff)
+				tt.Errorf("got:\n%s\nwant:\n%s\ndiff:\n%s", got, want, diff)
 			}
 		})
 	}

